packages/glob/src/internal-path-helper.ts

```diff
@@ -0,0 +1,144 @@
+import * as assert from 'assert'
+import * as path from 'path'
+
+const IS_WINDOWS = process.platform === 'win32'
+
```

Comment:
Why not use the `path` methods here instead?

There's `dirname`, `isAbsolute`, and `normalize` that have similar purposes to what's implemented here

Reply:
`dirname` and `isAbsolute` have quirks and for what i'm doing i need very precise behavior on Windows.

For example `dirname` usually removes unnecessary trailing slashes, but doesn't always for UNC paths (doesn't for `\\server\share\foo` but does for `\\server\share\`)

`isAbsolute` correctly returns false for `C:`. Whereas I need the answer to a different question: is it rooted? (yes for `C:`)

`normalize` also removes relative pathing. That's not the behavior I desire, i just want slashes normalized.